indexer/feature_utils.cpp

```diff
@@ -1,15 +1,78 @@
-#include "indexer/feature_utils.hpp"
-#include "indexer/feature_visibility.hpp"
 #include "indexer/classificator.hpp"
+#include "indexer/feature.hpp"
 #include "indexer/feature_data.hpp"
+#include "indexer/feature_utils.hpp"
+#include "indexer/feature_visibility.hpp"
 #include "indexer/scales.hpp"
 
 #include "geometry/point2d.hpp"
 
+#include "platform/preferred_languages.hpp"
+
+#include "coding/multilang_utf8_string.hpp"
+
 #include "base/base.hpp"
 
 #include "std/vector.hpp"
 
+namespace
+{
+struct BestMatchedLangNames
+{
+  string m_defaultName;
+  string m_nativeName;
+  string m_intName;
+  string m_englishName;
+
+  bool operator()(int8_t code, string const & name)
+  {
+    auto constexpr defaultCode = StringUtf8Multilang::kDefaultCode;
```

Comment:
Serious overhead.

09 авг. 2016 г. 16:33 пользователь "Arsentiy Milchakov" <
notifications@github.com> написал:

> In indexer/feature_utils.cpp
> <https://github.com/mapsme/omim/pull/3966#discussion_r74057778>:
>
> >  #include "base/base.hpp"
> >
> >  #include "std/vector.hpp"
> >
> > +namespace
> > +{
> > +struct BestMatchedLangNames
> > +{
> > +  string m_defaultName;
> > +  string m_nativeName;
> > +  string m_intName;
> > +  string m_englishName;
> > +
> > +  bool operator()(int8_t code, string const & name)
> > +  {
> > +    auto constexpr defaultCode = StringUtf8Multilang::kDefaultCode;
>
> да, ведь значения известны еще на этапе компиляции, а без constexpr
> переменные будут каждый раз инициализироваться
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/3966/files/23f281f4eb21cc6b38d649af9415ab03455b5ec7#r74057778>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAbvp065zd1hnKhxt_ixCBmNykbl5k7Qks5qeIGOgaJpZM4JdpwV>
> .
>